Refactor Kubernetes Executor Types to NamedTuples

HsiuChuanHsu opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->
### Description
When working on #54115, a good suggestion from @HTErik that we should refactors the Kubernetes executor type to make for better readability. 
This PR refactors the Kubernetes executor type system by converting simple tuple aliases to structured `NamedTuple` classes.

### Changes Made
- Type System Improvements
  - `KubernetesResultsType` → `KubernetesResults` (NamedTuple)
  - `KubernetesWatchType` → `KubernetesWatch` (NamedTuple)
  - `KubernetesJobType` → `KubernetesJob` (NamedTuple)

- Code Updates
  - Updated `kubernetes_executor_types.py`: Define new NamedTuple classes with proper typing
  - Updated `kubernetes_executor_utils.py`: Modify usage patterns to work with NamedTuples
  - Updated `kubernetes_executor.py`: Refactor result handling logic for new type structure
  - Updated `test_kubernetes_executor.py`: Adjust test assertions to work with named fields



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

HsiuChuanHsu committed:
refactor: Convert Kubernetes executor tuple types to NamedTuple
Replace complex tuple type aliases with NamedTuple classes for better
readability and maintainability:
- KubernetesResults: key, state, pod_name, namespace, resource_version, failure_details
- KubernetesWatch: pod_name, namespace, state, annotations, resource_version, failure_details
- KubernetesJob: key, command, kube_executor_config, pod_template_file

HsiuChuanHsu committed:
refactor(kubernetes): convert tuple types to NamedTuple for better readability
Replace tuple type aliases with structured NamedTuple classes in Kubernetes
executor to improve code maintainability and type safety.

hterik commented on providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/executors/kubernetes_executor_types.py:
```diff
@@ -35,27 +42,40 @@ class FailureDetails(TypedDict, total=False):
 
-    from airflow.models.taskinstance import TaskInstanceKey
-    from airflow.utils.state import TaskInstanceState
+    key: TaskInstanceKey  # 使用字符串引用
```
Please stay with English.

HsiuChuanHsu replied:
Forgot to remove the note. Thanks for the reminder!

hterik commented on providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py:
```diff
@@ -735,18 +738,26 @@ def _flush_result_queue(self) -> None:
+                            results.pod_name,
+                            results.namespace,
+                            results.failure_details,
+                        )
```
Would it make sense to change the signature of `_change_state` to accept only a single `result` argument?

HsiuChuanHsu replied:
Make sense! 
Simplify the `_change_state` method's signature to a single KubernetesResults object. This change will make the code much easier to use and read.

hterik reviewed (commented):
Nice. :+1: 
LGTM

HsiuChuanHsu committed:
refactor(kubernetes): simplify _change_state method to accept KubernetesResults object

- Simplify _change_state method signature from 5 parameters to 1 KubernetesResults object
- Update method calls in sync() and _flush_result_queue() to pass KubernetesResults directly
- Update all test cases to create KubernetesResults objects instead of passing individual parameters

uranusjr reviewed (approved):
Makes sense

HsiuChuanHsu committed:
fix(kubernetes): update integration tests to use KubernetesResults object

Update test_pod_failure_logging_with_container_terminated to create KubernetesResults object
Update test_pod_failure_logging_exception_handling to use new _change_state signature
Update test_pod_failure_logging_non_failed_state to use KubernetesResults object

- Remove unused FailureDetails import
- Format KubernetesResults constructor calls for better readability

jason810496 reviewed (commented):
Nice! Thanks for the PR!
LGTM if CI run is successful.

Unfortunately, linting still needs to be fixed in this run. It can be automatically resolved by using `prek` (the alternative to `pre-commit` we recently switched to) with the following command.

```bash
prek ruff --all-files && prek run ruff-format --all-files
```

HsiuChuanHsu committed:
Remove unused FailureDetails import from kubernetes_executor.py
